profiling/src/profiling/mod.rs

```diff
@@ -665,6 +665,54 @@ impl Profiler {
         }
     }
 
+    #[cfg(feature = "timeline")]
+    pub unsafe fn collect_compile_file(
+        &self,
+        duration: i64,
+        filename: String,
+        include_type: &str,
+        locals: &RequestLocals,
+    ) {
+        let mut labels = Profiler::message_labels();
+
+        lazy_static! {
+            static ref TIMELINE_COMPILE_FILE_LABELS: Vec<Label> = vec![Label {
+                key: "event",
+                value: LabelValue::Str("compilation".into()),
+            },];
+        }
+
+        labels.extend_from_slice(&TIMELINE_COMPILE_FILE_LABELS);
+        labels.push(Label {
+            key: "filename",
+            value: LabelValue::Str(Cow::from(filename)),
+        });
+        let n_labels = labels.len();
+
+        match self.send_sample(Profiler::prepare_sample_message(
```

Comment:
I realized just now that there's a decent number of lines of code between the measuring of the elapsed time and preparing the sample message, which is the timestamp. Do you think we should push the system time to the caller and pass it around?

Reply:
Thats a good idea 👍

Reply:
I'll created a task for this (PROF-7844) and will start working on it after this is merged.

Reply:
Sounds good! Remember to take the sample time after the event's duration has measured (timestamps match the end of the event, not the beginning).